### PR TITLE
Add support for network mode

### PR DESCRIPTION
**Pull request by ptiredknees**
Allow configuration of network mode which enables attaching test containers to an existing network.

### CODE REVIEW

**Comment by ptiredknees** [comment]
As an alternative, we could make `io.micronaut.testresources.testcontainers.TestContainers` a little smarter so that it uses an existing network if found.

Something like this:

```java
private static Optional<Network> findNamedNetwork(String name) {
    ListNetworksCmd listNetworksCmd = DockerClientFactory.instance()
        .client()
        .listNetworksCmd()
        .withNameFilter(name);
  
    return listNetworksCmd.exec().stream().findFirst()
        .map(network -> new Network() {
            @NotNull
            @Override
            public Statement apply(
                @NotNull Statement base,
                @NotNull Description description
            ) {
                return base;
            }
  
            @Override
            public String getId() {
                return network.getId();
            }
  
            @Override
            public void close() {
                // no-op
            }
        });
}

public static Network network(String name) {
      return NETWORKS_BY_KEY.computeIfAbsent(name, k ->
          findNamedNetwork(name).orElseGet(Network::newNetwork)
      );
}
```

File: src/main/docs/guide/modules-testcontainers.adoc [hunk 1]
```diff
@@ -161,6 +161,8 @@ test-resources:
 The `network` key is used to declare the network containers use.
 The `network-aliases` can be used to assign host names to the containers.
 
+The `network-mode` key can be used to join an existing network.
```
Comment (melix) [inline src/main/docs/guide/modules-testcontainers.adoc:164]: :x: I don't think this documentation is correct. In order to join an existing network, you need to use the `network` property. This is about the network _mode_ (bridge, etc...). I would rephrase the docs.

**Review by melix — CHANGES_REQUESTED** [review]
LGTM, except for the documentation part.

File: src/main/docs/guide/modules-testcontainers.adoc (hunk 1 above)
Comment (ptiredknees) [reply src/main/docs/guide/modules-testcontainers.adoc:164]: Well, for my use case, I provide the name of an existing bridge network I want my containers to join.
I agree that this is confusing which is why I suggested making a change to the existing `network` functionality.
What do you think?

Comment (melix) [reply src/main/docs/guide/modules-testcontainers.adoc:164]: But I think it's more confusing as it's not aligned with the testcontainers terminology. I think this should be rephrased to:

> The `network-mode` can be used to set the network mode (e.g. including 'host', 'bridge', 'none' or the name of an existing named network.)

Which is basically copied from the `withNetworkMode` javadocs.

**Review by melix — APPROVED** [review]

**Comment by melix** [comment]
There's one test failing, didn't take a look yet but it's related to these changes

**Review by melix — APPROVED** [review]

**Comment by melix** [comment]
Thank you!
